scipy/stats/morestats.py

```diff
@@ -1167,10 +1216,10 @@ def _optimizer(func, args):
         if brack is not None:
             raise ValueError("`brack` must be None if `optimizer` is given")
 
-        # `optimizer` is expected to return a `OptimizeResult` object, we here get the
-        # solution to the optimization problem.
+        # `optimizer` is expected to return a `OptimizeResult` object, we here
+        # get the solution to the optimization problem.
         def _optimizer(func, args):
-            return optimizer(func, args).x.item()
+            return getattr(optimizer(func, args=args), 'x', None)
```

Comment:
Was `.item()` really necessary?
If so, consider using `x[()]` instead. It is a bit more robust. It does the right thing when `x` is a 0D array, but it also does the right thing in some other cases.

Reply:
Unless we use `method="all"` it should be a float, whereas the `OptimizeResult` object stores `x` in an array, according to the documentation. So I'd think we have to return the scalar here.

Reply:
Hmm. Well it doesn't seem to be converting it to an array. As the code is now, `lmbda` is a `np.float64`, not an `ndarray`. Please try it out, and if you think it's OK, go ahead and merge.

Reply:
Yes seems to work without any problems.